test/e2e/aks.go

```diff
@@ -78,20 +82,15 @@ func DiscoverAndWaitForAKSControlPlaneInitialized(ctx context.Context, input Dis
 	Expect(input.Lister).NotTo(BeNil(), "Invalid argument. input.Lister can't be nil when calling DiscoverAndWaitForAKSControlPlaneInitialized")
 	Expect(input.Cluster).NotTo(BeNil(), "Invalid argument. input.Cluster can't be nil when calling DiscoverAndWaitForAKSControlPlaneInitialized")
 
-	controlPlane := GetAzureManagedControlPlaneByCluster(ctx, GetAzureManagedControlPlaneByClusterInput{
```

Comment:
Since we only ever use the Name and Namespace from this `controlPlane`, I opted to pull those from the Cluster instead so we don't need an ASOAPI-typed equivalent for this `GetAzureManagedControlPlaneByCluster` function which is otherwise unused.